sdk/keyvault/keyvault-secrets/src/lro/delete/operation.ts

```diff
@@ -0,0 +1,101 @@
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { RequestOptionsBase } from "@azure/core-http";
+import { DeletedSecret, SecretClientInterface } from "../../secretsModels";
+
+/**
+ * @interface
+ * An interface representing the state of a delete secret's poll operation
+ */
+export interface DeleteSecretPollOperationState extends PollOperationState<DeletedSecret> {
+  name: string;
+  requestOptions?: RequestOptionsBase;
+  client: SecretClientInterface;
+}
+
+/**
+ * @interface
+ * An interface representing a delete secret's poll operation
+ */
+export interface DeleteSecretPollOperation
+  extends PollOperation<DeleteSecretPollOperationState, DeletedSecret> {}
+
+/**
+ * @summary Reaches to the service and updates the delete secret's poll operation.
+ * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+ */
+async function update(
+  this: DeleteSecretPollOperation,
+  options: {
+    abortSignal?: AbortSignalLike;
+    fireProgress?: (state: DeleteSecretPollOperationState) => void;
+  } = {}
+): Promise<DeleteSecretPollOperation> {
+  const state = this.state;
+  const { name, client } = state;
+
+  const requestOptions = state.requestOptions || {};
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  if (!state.started) {
+    const deletedSecret = await client.deleteSecret(name, requestOptions);
+    state.started = true;
+    if (!deletedSecret.properties.recoveryId) {
+      state.completed = true;
+      state.result = deletedSecret;
```

Comment:
As with .NET and Python, I'd recommend setting the result anyway since the server already returned it and people may not need to wait to get the result. Depends on what they want to do next.